src/transformers/modeling_utils.py

```diff
@@ -5203,9 +5203,10 @@ def _assign_original_dtype(module):
                     "Kernels are not available. To use kernels, please install kernels using `pip install kernels`"
                 )
 
-            from kernels import Device, kernelize
+            from kernels import Device, Mode, kernelize
 
-            kernelize(model, device=Device(type=model.device.type))
+            mode = Mode.INFERENCE if not model.training else Mode.TRAINING
```

Comment:
```suggestion
            mode = Mode.TRAINING if model.training else Mode.INFERENCE
```

(ultranit, completely ignorable; it flows slightly better without the `not` imo)